assets/src/edit-story/components/panels/layer/useLayerReordering.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import {
+  useState,
+  useCallback,
+  useEffect,
+  useContext,
+  useRef,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app';
+import LayerContext from './context';
```

Comment:
Sortable component? As in a generalized approach?

Reply:
Yes, the `react-sortable-hoc` component that came up. I thought you said you'll take a look if it's applicable here and maybe for page carousel/grid as well.

Reply:
Ah, `react-sortable-hoc` uses a completely different approach to both mouse-based and keyboard-based interaction than we need here.

We don't actually need anything to drag (we just show separators while the mouse is down), and we don't need any keyboard interaction at all.

And if we did need keyboard interaction, it would not be the model used by `react-sortable-hoc`, which is _space to grab, arrows to move, space to release_.

Reply:
Thanks! Do you anticipate though that we will need to refactor bulk of the pointer code here for page carousel? On the surface it seems more or less similar: pointerover+separator work. Does this model still work for the grid?

Reply:
Yes and no. I tried to generalize with carousel and grid in mind, but ending up not worrying about it for now. Because for both carousel and grid we'll need keyboard accessibility to reorder as well. So we can use the same approach, but would need to add this on top somehow.

Reply:
Opened #332 for carousel and grid view.